demosplan/DemosPlanCoreBundle/Addon/AddonResolveTargetEntity.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of the package demosplan.
+ *
+ * (c) 2010-present DEMOS plan GmbH, for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+namespace demosplan\DemosPlanCoreBundle\Addon;
+
+use demosplan\DemosPlanCoreBundle\Utilities\DemosPlanPath;
+use RecursiveDirectoryIterator;
+use RecursiveIteratorIterator;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+
+/**
+ * Handles connecting AddOn interfaces with their corresponding CoreEntities
+ */
+class AddonResolveTargetEntity implements CompilerPassInterface
+{
+
+    private const CORE_ENTITY_DIRECTORY = 'demosplan/DemosPlanCoreBundle/Entity';
+    private const ADDON_INTERFACE_DIRECTORY = 'DemosEurope\DemosplanAddon\Contracts\Entities';
+
+    public function process(ContainerBuilder $container): void
+    {
+        $definition = $container->findDefinition('doctrine.orm.listeners.resolve_target_entity');
+        $corePath = DemosPlanPath::getRootPath(self::CORE_ENTITY_DIRECTORY);
+
+
+        $iterator = new RecursiveDirectoryIterator($corePath);
```

Comment:
I like to use the `Filesystem` Component from Symfony for such tasks, but this is just as I like it

Reply:
Thx for the feedback @mussbach I will use it next time